From c3e8164ef18a4cceafeb97a85f6924172187e44d Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Wed, 18 May 2022 15:49:13 +0200
Subject: [PATCH 07/20] onlpm: add an argument for just printing the builddirs

To allow building without packaging, we need to know the appropriate
build dir for passing to make.

Add an argument to onlpm that allows printing the appropriate build
directories of packages.

Upstream-Status: Inappropriate [Hack for building with yocto]
Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 tools/onlpm.py | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/tools/onlpm.py b/tools/onlpm.py
index d4403c0d3b65..e810ab7ec76a 100755
--- a/tools/onlpm.py
+++ b/tools/onlpm.py
@@ -1297,6 +1297,7 @@ if __name__ == '__main__':
     ap.add_argument("--in-repo", nargs='+', metavar='PACKAGE')
     ap.add_argument("--include-env-json", default=os.environ.get('ONLPM_OPTION_INCLUDE_ENV_JSON', None))
     ap.add_argument("--platform-manifest", metavar=('PACKAGE'))
+    ap.add_argument("--show-build-dirs", nargs='+', metavar='PACKAGE')
 
     ops = ap.parse_args()
 
@@ -1416,6 +1417,15 @@ if __name__ == '__main__':
                 else:
                     raise OnlPackageMissingError(p)
 
+        if ops.show_build_dirs:
+            for p in ops.show_build_dirs:
+                if p in pm:
+                    for pg in pm.filtered_package_groups():
+                        if p in pg:
+                            print(pg._pkgs['__directory'])
+                else:
+                    raise OnlPackageMissingError(p)
+
         if ops.require:
             for p in ops.require:
                 pm.require(p, force=ops.force, build_missing=not ops.no_build_missing,
-- 
2.51.0

